native/c/zds.cpp

```diff
@@ -304,6 +304,92 @@ int zds_write_to_dsn(ZDS *zds, const string &dsn, string &data)
   return 0;
 }
 
+int zds_open_output_bpam(ZDS *zds, std::string dsname, IO_CTRL *&ioc)
+{
+  string alloc_cmd = "ALLOC DA('" + dsname + "') SHR"; // TODO(Kelosky): log this command
+  unsigned int code = 0;
+  string resp = "";
+  string ddname = "        "; // set to empty string to request a dynamic DD name
+  int rc = zut_bpxwdyn(alloc_cmd, &code, resp, ddname);
+  if (0 != rc)
+  {
+    strcpy(zds->diag.service_name, "BPXWDYN");
+    zds->diag.e_msg_len = sprintf(zds->diag.e_msg, "Failed to allocate with code '%08X' on data set '%s': %s", code, dsname.c_str(), resp.c_str());
+    zds->diag.detail_rc = ZDS_RTNCD_SERVICE_FAILURE;
+    zds->diag.service_rc = code;
+    return RTNCD_FAILURE;
+  }
+
+  zds->dynalloc = 1;
+  zut_uppercase_pad_truncate(zds->ddname, ddname.c_str(), sizeof(zds->ddname));
+
+  rc = ZDSOBPAM(zds, &ioc, zds->ddname);
+  if (0 != rc)
+  {
+    return rc;
+  }
+  return RTNCD_SUCCESS;
+}
+
+int zds_write_output_bpam(ZDS *zds, IO_CTRL *ioc, string &data)
```

Comment:
Consider implementing an overload to support passing a `const char*` and length to avoid the overhead of creating a `std::string` instance for each chunk (in the case of streaming)